src/odemis/dataio/tiff.py

```diff
@@ -1243,6 +1251,11 @@ def _addImageElement(root, das, ifd, rois, fname=None, fuuid=None):
             for j in range(3):
                 trane.attrib["A%d%d" % (i, j)] = "%.15f" % trans_mat[i][j]
 
+    if model.MD_EXTRA_SETTINGS in globalMD:
+        sett = globalMD[model.MD_EXTRA_SETTINGS]
+        extrase = ET.SubElement(ime, "ExtraSettings")
+        extrase.text = json.dumps(sett)  # serialize hw settings
```

Comment:
For safety, I'd suggest to protect the `dumps()` in a try/except. So that if there is some weird object in the dictionary, it doesn't prevent from entirely saving the data.